core/src/test/java/com/sap/ai/sdk/core/AiCoreServiceTest.java

```diff
@@ -0,0 +1,143 @@
+    final var core = new AiCoreService();
+    final var destination = core.destination();
+    final var client = core.client();
+
+    // verification
+    assertThat(destination.get(DestinationProperty.URI)).contains("https://srv/v2/");
+    assertThat(destination.get(DestinationProperty.AUTH_TYPE)).isEmpty();
+    assertThat(destination.get(DestinationProperty.NAME)).singleElement(STRING).contains("aicore");
+    assertThat(destination.get(AI_CLIENT_TYPE_KEY)).contains(AI_CLIENT_TYPE_VALUE);
+    assertThat(client.getBasePath()).isEqualTo("https://srv/v2/");
+    verify(accessor, times(2)).getServiceBindings();
+  }
+
+  @Test
+  void testBaseDestination() {
+    // setup
+    DestinationResolver.setAccessor(Collections::emptyList);
+
+    // execution without errors
+    final var customDestination = DefaultHttpDestination.builder("https://foo.bar").build();
+    final var core = new AiCoreService().withDestination(customDestination);
+    final var destination = core.destination();
+    final var client = core.client();
+
+    // verification
+    assertThat(destination.get(DestinationProperty.URI)).contains("https://foo.bar/v2/");
+    assertThat(destination.get(DestinationProperty.AUTH_TYPE)).isEmpty();
+    assertThat(destination.get(DestinationProperty.NAME)).isEmpty();
+    assertThat(destination.get(AI_CLIENT_TYPE_KEY)).contains(AI_CLIENT_TYPE_VALUE);
+    assertThat(client.getBasePath()).isEqualTo("https://foo.bar/v2/");
+  }
+
+  @Test
+  void testDeployment() {
+    final var accessor = mock(ServiceBindingAccessor.class);
+    DestinationResolver.setAccessor(accessor);
+    doReturn(List.of(BINDING)).when(accessor).getServiceBindings();
+
+    // execution without errors
+    final var destination = new AiCoreService().destination();
+    final var client = new AiCoreService().client();
+
+    // verification
+    assertThat(destination.get(DestinationProperty.URI)).contains("https://srv/v2/");
+    assertThat(destination.get(DestinationProperty.AUTH_TYPE)).isEmpty();
+    assertThat(destination.get(DestinationProperty.NAME)).singleElement(STRING).contains("aicore");
+    assertThat(destination.get(AI_CLIENT_TYPE_KEY)).contains(AI_CLIENT_TYPE_VALUE);
+    assertThat(client.getBasePath()).isEqualTo("https://srv/v2/");
+    verify(accessor, times(2)).getServiceBindings();
+  }
+
+  @Test
+  void testCustomization() {
+    final var customService =
+        new AiCoreService() {
+          @Nonnull
+          @Override
+          protected Destination getBaseDestination() {
+            return DefaultHttpDestination.builder("https://ai").build();
+          }
+
+          @Nonnull
+          @Override
+          protected ApiClient getApiClient(@Nonnull Destination destination) {
+            return new ApiClient().setBasePath("https://fizz.buzz").setUserAgent("SAP");
+          }
+        };
+
+    final var customServiceForDeployment =
+        customService.forDeployment("deployment").withResourceGroup("group");
```

Comment:
verify that `withResourceGroup` works

Reply:
Added test assertion here
https://github.com/SAP/ai-sdk-java/pull/79/commits/957510986a5d5ef499d4fcc586a60e193a4e07ae